webhook/main.py

```diff
@@ -1,7 +1,23 @@
-from fastapi import FastAPI, Request
+from fastapi import FastAPI, Request, Header, HTTPException, Body
+from pydantic import BaseModel
+from typing import List, Optional, Dict, Union
 
 app = FastAPI()
 
+
 @app.get("/")
 async def root():
     return {"data": "Hello World"}
+
+
+class VTAPI(BaseModel):
+  data: Union[str, List, Dict]
```

Comment:
A response from VT API's `/api/v3/intelligence/search` has the following structure:

```
{
  "data": [],
  "links": {
    "next": "",
    "self": ""
  },
  "meta": {}
}
```

Every item in the `data` list has the following structure:
```
{
  "attributes": {},
  "id": "",
  "links": {"self": ""},
  "type": "this can be either file, url, domain, ip_address"
}
```

So instead of defining data as either a str, list or dictionary. I'd have a separate class:

```python
class VTObjects(str, Enum):
  file = 'file'
  url = 'url'
  domain = 'domain'
  ip_address = 'ip_address'

class VTData(BaseModel):
  attributes: Dict
  id: str
  links: Dict
  type: VTObjects
```

And finally, I'd define data as:
```suggestion
  data: VTData
```

Reply:
https://developers.virustotal.com/v3.0/reference#api-responses

> <response data> is usually an object or a list of objects, but that's not always the case. An example of this is the /files/upload_url endpoint, which returns a URL. Refer to each endpoint's documentation to learn more about its return data structure.

Why do you assume, that `"data": [],` will always be a list?

Why "next": "", is not optional? I tried to do some requests to VT yesterday and I'm not sure that links are obligatory (for example my query result didn't have "next" in "links".

Reply:
It will always be a list, because you will just receive responses from the `/api/v3/intelligence/search` endpoint

Reply:
next is optional, yes

Reply:
Done